test/helpers/screenshot_test.exs

```diff
@@ -4,10 +4,73 @@ defmodule ScreenshotTest do
 
   hound_session()
 
-  test "should take a screenshot" do
+  setup do
     navigate_to("http://localhost:9090/page1.html")
-    path = take_screenshot("screenshot-test.png")
+    :ok
+  end
+
+  test "should take a screenshot with a generated file name" do
+    default_path = File.cwd!()
+
+    path = take_screenshot()
+
     assert File.exists?(path)
+    assert path =~ default_path
+    assert file_name(path) =~ ~r/^screenshot-\d{4}(?:-\d{1,2}){5}\.png$/
   end
 
+  test "file names generated 1 second apart are unique" do
+    default_path = File.cwd!()
+
+    path1 = take_screenshot()
+    :timer.sleep(1000)
+    path2 = take_screenshot()
+
+    paths = [path1, path2]
+    assert paths == Enum.uniq(paths)
+  end
+
+  describe "When given a file name" do
+    test "should take a screenshot at the default location with the specified file name" do
+      default_path = File.cwd!()
+
+      take_screenshot("screenshot-test.png")
+
+      assert File.exists?("#{default_path}/screenshot-test.png")
+    end
+  end
+
+  describe "When given a file name with path" do
+    test "should take a screenshot at the specified location with the specified file name" do
+      take_screenshot("test/screenshot-test.png")
+
+      assert File.exists?("test/screenshot-test.png")
+    end
+  end
+
+  describe "When :screenshot_dir is set" do
+    setup do
+      screenshot_dir = "tmp/screenshots/"
+      Application.put_env(:hound, :screenshot_dir, screenshot_dir)
+      change_session_to(:with_opts, [screenshot_dir: screenshot_dir])
```

Comment:
These seem to happen to late in the application lifecycle. I'm not sure if there's an earlier even I can hook into?

Reply:
This will normally be configured by `config.exs` anyway, so let's stick with what you have here.

Reply:
@tuvistavie sorry, should have been more clear. Neither of the options (line 54 or 55 above) actually change the screenshot directory in the test so all it's currently testing is that the file name is generated in the same way.

This line actually fails right now:
https://github.com/marcdel/hound/blob/9a627eab709e5b852ee08eb2934e1fc37296128b/test/helpers/screenshot_test.exs#L66